drivers/mulesoft/calendar_exporter.cr

```diff
@@ -72,9 +74,8 @@ class MuleSoft::CalendarExporter < PlaceOS::Driver
   end
 
   protected def export_booking(booking : Hash(String, Int64 | String | Nil))
-    # Mulesoft booking titles are often nil. Use the body instead in this case
-    booking["title"] = booking["body"] if booking["title"].nil?
-    booking["title"] = "#{booking["recurring_master_id"]} #{booking["title"]}"
+    # Add the course code to the front of the booking title/body
+    booking["title"] = "#{booking["recurring_master_id"]} #{booking["title"] || booking["body"]}"
```

Comment:
Is there a chance that this method could be called twice for the same booking?

Reply:
I believe export_booking will run once for each item in @bookings, and only AFTER @bookings is successfully updated with the (real) booking data from the Mulesoft module.

So our local booking["title"] gets transformed but then replaced with Mulesoft (no transformed) data before export_booking is run again.

If @bookings fails to get replaced with the Mulesoft bookings then I think there would be an error before hitting `@bookings.each { |b| export_booking(b) }` on line 43

Reply:
My worry is that this method call won't be idempotent (different if run again with the same inputs), but following the code it seems okay to me.

Reply:
yeah makes sense, is the best thing to do usually to clone the object before transforming, like this?

Reply:
i've made a change where a booking clone is transformed instead of the booking in the instance variable. have requested new review